runner/src/server/plugins/engine/pageControllers/RepeatingSummaryPageController.ts

```diff
@@ -103,11 +125,34 @@ export class RepeatingSummaryPageController extends PageController {
         },
       };
     });
+    return rows;
+  }
 
-    return {
-      ...baseViewModel,
-      details: { rows },
-    };
+  buildTextFieldRows(formData, input) {
+    delete formData["progress"];
+    const rows = Object.keys(formData[input.name]).map((key) => {
+      const titleWithIteration = `${
+        formData[input.name][key]["type-of-revenue-cost"]
```

Comment:
Should this key be hardcoded?

Reply:
For now yes, its something that I really dont like, but the values for these parts of the array a hardcoded within the class structure